cmd/scaffold.go

```diff
@@ -561,6 +563,23 @@ func (fnb *FlowNodeBuilder) initSecretsDB() {
 	})
 }
 
+// loadSecretsEncryptionKey attempt to read an encryption key for the secrets DB from the canonical path
+func (fnb *FlowNodeBuilder) loadSecretsEncryptionKey() []byte {
+	encryptionKey, err := loadSecretsEncryptionKey(fnb.BootstrapDir, fnb.NodeID)
+	if errors.Is(err, os.ErrNotExist) {
+		if fnb.NodeRole == flow.RoleConsensus.String() {
```

Comment:
The logic of the additional constraint on consensus nodes is split between this function and `initSecretsDB` currently. To fully understand the logic for this we need to read both functions `loadSecretsEncryptionKey`.

We could make this function agnostic to the role-specific constraints (like the other functions which read bootstrap files, which just read and return the file contents) and consolidate the special logic for consensus nodes into `initSecretsDB`, to make it a bit easier to follow.

Eg.
```go
func loadSecretsEncryptionKey() ([]byte key, bool fileExists, err error) { ... }
```

Reply:
Good catch, I left all the logic in initSecretsDB theres not much need to add separate func at the moment https://github.com/onflow/flow-go/pull/2042/commits/29136d315045e543a9be3631b7e51b774f8ff689